src/Compilers/Core/Portable/MetadataReader/PEModule.cs

```diff
@@ -1134,6 +1134,47 @@ internal bool HasIsByRefLikeAttribute(EntityHandle token)
             return null;
         }
 
+#nullable enable
+        internal UnmanagedCallersOnlyAttributeData? TryGetUnmanagedCallersOnlyAttribute(
+            EntityHandle token,
+            IAttributeNamedArgumentDecoder attributeArgumentDecoder,
+            Func<string, TypedConstant, (bool IsCallConvs, ImmutableHashSet<INamedTypeSymbolInternal>? CallConvs, bool IsValid)> unmanagedCallersOnlyDecoder)
+        {
+            AttributeInfo info = FindTargetAttribute(token, AttributeDescription.UnmanagedCallersOnlyAttribute);
+            if (!info.HasValue || info.SignatureIndex != 0 || !TryGetAttributeReader(info.Handle, out BlobReader sigReader))
+            {
+                return null;
+            }
+
+            var unmanagedConventionTypes = ImmutableHashSet<INamedTypeSymbolInternal>.Empty;
+            bool isValid = true;
+
+            if (sigReader.RemainingBytes > 0)
+            {
+                try
+                {
+                    var numNamed = sigReader.ReadUInt16();
+                    for (int i = 0; i < numNamed; i++)
+                    {
+                        var ((name, value), _, _) = attributeArgumentDecoder.DecodeCustomAttributeNamedArgumentOrThrow(ref sigReader);
```

Comment:
>_, _ [](start = 44, length = 4)

Why are we ignoring these values?

Reply:
The first one is whether this is a property or not, I can check that. The second is the typeCode of the deserialized data, and that is checked by the main decode step itself.

---
In reply to: [493008750](https://github.com/dotnet/roslyn/pull/47950#discussion_r493008750) [](ancestors = 493008750)

Reply:
> The second is the typeCode of the deserialized data, and that is checked by the main decode step itself.

Consider adding a comment about this.

---
In reply to: [493888966](https://github.com/dotnet/roslyn/pull/47950#discussion_r493888966) [](ancestors = 493888966,493008750)

Reply:
Now actually checked. I added tests that will fail without these.

---
In reply to: [493918626](https://github.com/dotnet/roslyn/pull/47950#discussion_r493918626) [](ancestors = 493918626,493888966,493008750)